src/plugins/data_source/docs/user_stories.md

```diff
@@ -0,0 +1,73 @@
+
+This document discusses the user experience of the OpenSearch Dashboards multiple data source support.
+
+## User Story
+
+### Current user experience
+
+- OpenSearch Dashboards admin setup the OpenSearch Dashboards service and configure the OpenSearch endpoint in `opensearch_dashboards.yml`
+  - Both the OpenSearch Dashboards metadata index (`opensearch_dashboards` index) and data indices are saved in the same OpenSearch cluster
+- OpenSearch Dashboards users can work with visualizations, usually they will
+  - Create/update index patterns
+  - Create/update visualization, each visualization is built on top of one index pattern
+  - Create/update dashboard using a group of visualizations
+  - Run adhoc queries against an index pattern using discover feature
+  - View index patterns/visualization/dashboards
+- OpenSearch Dashboards users can work with analytics functions, such as Alerting/AD etc
+
+### Expected user experience with multiple data source
+
+We are planning to introduce a new `data-source` model, to describe an OpenSearch data source, and letting index pattern to refer to a `data-source`.
+
+- OpenSearch Dashboards admin setup the OpenSearch Dashboards service and configure the OpenSearch **metadata store endpoint** in `opensearch_dashboards.yml`
+  - the metadata store OpenSearch cluster only saves the `opensearch_dashboards` index, data indices can be saved in other OpenSearch stores
+- Users will need to have a data-source before they can do any visualization or analytics work with OpenSearch Dashboards
+  - Users can create/update/view data sources
+  - Users need to specify a data source when creating new index patterns, data source is not mutable after index pattern is created
+  - Create/update visualization and dashboards experience remains the same as is today.
+  - View index patterns/visualization/dashboards experience remains the same as is today.
+- When users want to work with analytics features like AD and alerting. they need to specify a data source to work with. (We may consider to add default data source concept)
+
+## UI Change
+
+This multiple data source support and introduction of data source model requires several UI changes on OpenSearch Dashboards
+
+### Data source management
+
+![img](./img/dsm_flow.png)
+
+Data source, as a new saved object type, should have a management page, like index pattern.
+
+We will need to
+
+- add a new data source entry in the stack management Nav app, with a data source list table
+- a data source detail page, to show detailed information of a specific data source, such as URL, auth type, endpoint capabilities etc.
+
+### Credential management
+
+![img](./img/cm_flow.png)
+
+Credential is used to establish an authenticated connection to other data source. Typical credentials are User/Password for basic auth. IAM auth for AWS specific authentication.
+
+Credential management is provided way for user to add/edit/remove the credential to be used in datasource management
```

Comment:
We removed credential management for security reason.

Reply:
Yeah, I'll remove this apart along with the diagram